### PR TITLE
Fixes the logout flow in htmx and improves the login error dialog

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #

Checklist

-  [ x ] This pr contains no AI generated code
- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: server/core/src/https/views/login.rs [hunk 1]
```diff
@@ -617,7 +658,7 @@ async fn view_login_step(
                             let mut username_cookie =
                                 Cookie::new(COOKIE_USERNAME, session_context.username.clone());
                             username_cookie.set_secure(state.secure_cookies);
-                            username_cookie.set_same_site(SameSite::Strict);
+                            username_cookie.set_same_site(SameSite::Lax);
```
Comment (yaleman) [inline server/core/src/https/views/login.rs:661]: I clearly missed this on a previous one, why aren't we using `Strict` here (and above, which is scarier)?

Comment (Firstyear) [reply server/core/src/https/views/login.rs:661]: https://developer.mozilla.org/en-US/docs/Web/Security/Practical_implementation_guides/Cookies#samesite

tl;dr is if it's strict then the cookie isn't sent on a redirect to kanidm. The redirector can't see the cookie, but lax allows it to be added on that redirection.

Comment (yaleman) [reply server/core/src/https/views/login.rs:661]: Ah, 'k. With the other protections in place that'll work 😄 👍🏻

File: server/core/src/https/views/apps.rs
Comment (ToxicMushroom) [inline server/core/src/https/views/apps.rs:1]: These changes make me wonder if my cargo fmt really works
My toolchain is set to stable with `rustc 1.79.0`

Comment (Firstyear) [reply server/core/src/https/views/apps.rs:1]: Yeah, something seems off, we have to keep reformating after you ..... :(

**Comment by Firstyear** [comment]
Thanks @ToxicMushroom those were good points. I've fixed them all now :)

**Review by ToxicMushroom — APPROVED** [review]

**Review by yaleman — APPROVED** [review]
